spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/SpanSubscriber.java

```diff
@@ -94,15 +101,15 @@ public void request(long n) {
 		if (log.isTraceEnabled()) {
 			log.trace("Request");
 		}
-		try (Tracer.SpanInScope ws = this.tracer.withSpanInScope(this.span)) {
```

Comment:
if the span was already in scope, this would have synced MDC etc again for no reason